analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.py

```diff
@@ -0,0 +1,125 @@
+# Author Teja Koganti (D3B)
+
+import sys
+
+import argparse
+import pandas as pd
+import numpy as np 
+import zipfile
+import statistics
+import  scipy
+from scipy import stats
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--notebook', required = True,
+                    help = 'path to the notebook file')
+parser.add_argument('--gistic', required = True,
+                    help = "gistic zip file")
+parser.add_argument('--subfile-gistic', required = True,
+                    help = "subfile of the zip folder that contains broad values by arm")
+parser.add_argument('--gsva', required = True,
+                    help = "gsva scores file")
+parser.add_argument('--expression', required = True,
+                    help = "expression subset file")
+parser.add_argument('--fusion', required = True,
+                    help = "fusion results file")
+parser.add_argument('--breakpoints', required = True,
+                    help = "breakpoint summary file")
+parser.add_argument('-o', '--outfile', required = True,
+                    help = "output file")
+args = parser.parse_args()
+
+# File to write all the data in 
+outfile = open(args.outfile, "w")
+
+# Reading GISTIC broad_values file for CNA 
+zip=zipfile.ZipFile(args.gistic)
+CNA=pd.read_csv(zip.open(args.subfile_gistic), sep="\t")
+CNA = CNA.set_index("Chromosome Arm")
+
+
+# Reading in gene set enrichment analyses file for GSEA scores for NKKB pathway
+gsva = pd.read_csv(args.gsva, sep="\t")
+gsva_NFKB = gsva.loc[gsva['hallmark_name'] == "HALLMARK_TNFA_SIGNALING_VIA_NFKB"]
+gsva_NFKB = gsva_NFKB.set_index("Kids_First_Biospecimen_ID")
+
+# Reading subset gene expression file
+fpkm_df = pd.read_csv(args.expression, sep = "\t")
+fpkm_df = fpkm_df.set_index("GENE")
+zscore_fpkm_df = fpkm_df.apply(scipy.stats.zscore)
+
+#Reading fusion summary file
+fusion = pd.read_csv(args.fusion, sep="\t")
+fusion = fusion.set_index("Kids_First_Biospecimen_ID")
+
+#Reading chromosomal instability file for breakpoint density 
+breakpoint_density = pd.read_csv(args.breakpoints, sep="\t")
+breakpoint_density = breakpoint_density.set_index("samples")
+
+#Reading the input in a  dataframe 
+EPN_notebook = pd.read_csv(args.notebook, sep="\t")
+
+# This function takes in CNA dataframe along  with chromosomal arm
+# and if should be a loss/gain in the results and returns a boolean value  accordingly
+def DNA_samples_fill_df(row, CNA, arm, loss_gain):
+    if row["Kids_First_Biospecimen_ID_DNA"] is np.nan:
+        return(0)
+    CNA_value = CNA.loc[arm, row["Kids_First_Biospecimen_ID_DNA"]]
+    if CNA_value<0 and loss_gain=="loss":
+       return(1)
+    elif loss_gain=="gain" and CNA_value>0:
+       return(1)
+    else:
+       return(0)
+
+# Function to generate Z-scores column for every gene 
+def fill_df_with_fpkm_zscores(df,fpkmdf, column_name, gene_name):
+        zscore_list = scipy.stats.zscore(np.array(df.apply(lambda x: fpkmdf.loc[gene_name, x["Kids_First_Biospecimen_ID_RNA"]], axis=1)))
```

Comment:
Using `stats` directly, as above.
```suggestion
        zscore_list = stats.zscore(np.array(df.apply(lambda x: fpkmdf.loc[gene_name, x["Kids_First_Biospecimen_ID_RNA"]], axis=1)))
```